### PR TITLE
chore(hybrid-cloud): Stable silo tests run by default

### PR DESCRIPTION
**Pull request by corps**
Ok, so here's a working example of my approach --

In the new world, we don't have to run the test suite in two modes.  We just run the test suite like we always do.  IFF the `SILO_MODE_UNSTABLE_TESTS` is set, then all tests marked with silo regions run both in monolith and in that silo, inline with the current invocation.   Otherwise, the tests all run with whatever silo mode was set in the environment variable, UNLESS the test is marked stable (in which case it runs with all the silo modes for that group!)

This would basically put our test suite on master, behind tests we mark as 'stable', helping us catch regressions.
We combine this with our tool to create the model-manifest, and write an additional 'linter' (just for us, not in everyone's environment) that allows us to quickly identify and mark all tests along a single model as 'stable'.  Then, we can run -just those tests- (with run unstable tests set in our person .env) until they all pass, lock it down, generate PR, close jira showing our progress to lockdown.

### CODE REVIEW

File: src/sentry/testutils/silo.py [hunk 1]
```diff
@@ -20,31 +21,25 @@ class SiloModeTest:
     By default, the test is executed if the environment is in that silo mode or
     in monolith mode. The test is skipped in an incompatible mode.
 
-    If the SILO_MODE_SPLICE_TESTS environment flag is set, any decorated test
-    class will be modified by having new test methods inserted. These new
-    methods run in the given modes and have generated names (such as
-    "test_response__in_region_silo"). This can be used in a dev environment to
-    test in multiple modes conveniently during a single test run. Individually
-    decorated methods and stand-alone functions are treated as normal.
+    Optionally, decorators constructed from this class can be provided the stable=True
```
Comment (corps) [inline src/sentry/testutils/silo.py:24]: I'll need to update this comment lol

File: src/sentry/testutils/silo.py [hunk 2]
```diff
@@ -20,31 +21,25 @@ class SiloModeTest:
...
     def __init__(self, *silo_modes: SiloMode) -> None:
         self.silo_modes = frozenset(silo_modes)
-        self.splice = bool(settings.SILO_MODE_SPLICE_TESTS)
+        self.run_unstable_tests = bool(settings.SILO_MODE_UNSTABLE_TESTS)
 
     @staticmethod
     def _find_all_test_methods(test_class: type) -> Iterable[Tuple[str, TestMethod]]:
         for attr_name in dir(test_class):
-            if attr_name.startswith("test_"):
+            if attr_name.startswith("test_") or attr_name == "test":
```
Comment (corps) [inline src/sentry/testutils/silo.py:36]: minor bug I noticed

File: bin/decorate-silo-mode-tests [hunk 3]
```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env python
+import ast
```
Comment (corps) [inline bin/decorate-silo-mode-tests:2]: I'm thinking that once we have the models-manifest, we can take that metadata and automatically tag all the tests associated with a model with the appropriate decorator, and set it to stable mode when ready.

Comment (corps) [reply bin/decorate-silo-mode-tests:2]: So this now works.  There's a natural workflow:

First, decide which silo mode the model should work in, and make sure all related tests are decorated consistently
`bin/decorate-silo-mode-tests UserIP --silo-mode control`

Second, `SILO_MODE_UNSTABLE_TESTS=1 bin/run-model-tests.py UserIP` will execute all tests in the manifest file associated with the UserIP model, and run them with silo tests activated.

Thirdly, make changes to get these tests passing!  For tests that are still failing in more fundamental ways (or are multi model tests), we can skip them for now.  Once the vast majority of model tests are passing, decorate them as stable: `bin/decorate-silo-mode-tests UserIP --silo-mode control --set-stable`

Fourthly, make PR.  Now all the UserIP model tests would run in monolith and silo mode by default!  Check the CI to ensure things pass and test markings make sense.  Remove any markings from tests that are unstable or still in need of fixing.

Fifth, merge and repeat process until all test models and their tests pass in CI.

**Review by RyanSkonnord — APPROVED** [review]

**Pull request merged**